connection.go

```diff
@@ -131,70 +152,25 @@ func (ct *ConnectionTransportTLS) Close() {
 	}
 }
 
-// SharedKeybaseTransport is a ConnectionTransport implementation that
-// uses a shared local socket to a keybase daemon.
-type SharedKeybaseTransport struct {
-	kbCtx *libkb.GlobalContext
-
-	// Protects everything below.
-	mutex           sync.Mutex
-	transport       rpc.Transporter
-	stagedTransport rpc.Transporter
-}
-
-// Test that SharedKeybaseTransport fully implements the
-// ConnectionTransport interface.
-var _ ConnectionTransport = (*SharedKeybaseTransport)(nil)
-
-// Dial is an implementation of the ConnectionTransport interface.
-func (kt *SharedKeybaseTransport) Dial(ctx context.Context) (
-	rpc.Transporter, error) {
-	_, transport, _, err := kt.kbCtx.GetSocket(true)
-	if err != nil {
-		return nil, err
-	}
-
-	kt.mutex.Lock()
-	defer kt.mutex.Unlock()
-	kt.stagedTransport = transport
-	return transport, nil
-}
-
-// IsConnected is an implementation of the ConnectionTransport interface.
-func (kt *SharedKeybaseTransport) IsConnected() bool {
-	kt.mutex.Lock()
-	defer kt.mutex.Unlock()
-	return kt.transport != nil && kt.transport.IsConnected()
-}
-
-// Finalize is an implementation of the ConnectionTransport interface.
-func (kt *SharedKeybaseTransport) Finalize() {
-	kt.mutex.Lock()
-	defer kt.mutex.Unlock()
-	kt.transport = kt.stagedTransport
-	kt.stagedTransport = nil
-}
-
-// Close is an implementation of the ConnectionTransport interface.
-func (kt *SharedKeybaseTransport) Close() {
-	// Since this is a shared connection, do nothing.
-}
+type LogTagsFromContext func(ctx context.Context) (map[interface{}]string, bool)
```

Comment:
hmm, 2 questions:
1) What was `SharedKeybaseTransport` doing that we lose now. I'm especially curious with respect to the `stagedTransport` vs `transport`. It's unclear to me what's happening there.
2) Why are `LogTags` `map[interface{}]string`, and not `map[string]interface{}` like [here](https://github.com/keybase/go-framed-msgpack-rpc/blob/1a388e533c8cd1e2eb4dc9dc70d6b70fb7cfb1a6/context.go#L33)